### PR TITLE
recipe2plan: New means of generating `Create` handle `StorageKey`s.

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Updating storage key generation: Now, if a handle has a `create` fate and a key has not been assigned in recipe resolution, we will generate a `CreateableStorageKey` from the handle id.

### CODE REVIEW

File: src/tools/tests/goldens/WriterReaderExample.kt [hunk 1]
```diff
@@ -28,7 +33,7 @@ object ConsumptionPlan : Plan(
             "",
             mapOf(
                 "data" to HandleConnection(
-                    StorageKeyParser.parse("reference-mode://{db://25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516@arcs/Thing}{db://25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516@arcs/!:writingArcId/handle/my-handle-id}"),
+                    StorageKeyParser.parse("db://25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516@arcs/!:writingArcId/handle/my-handle-id"),
```
Comment (mmandlis) [inline src/tools/tests/goldens/WriterReaderExample.kt]: why isn't it a reference-mode-storage key anymore?

File: src/tools/plan-generator.ts [hunk 2]
```diff
@@ -88,9 +88,12 @@ export class PlanGenerator {
     }
   }
 
-  /** Generates a Kotlin `StorageKey` from a StorageKey. */
-  createStorageKey(storageKey: StorageKey | undefined): string {
-    return `StorageKeyParser.parse("${(storageKey || '').toString()}")`;
+  /** Generates a Kotlin `StorageKey` from a recipe Handle. */
+  createStorageKey(handle: Handle): string {
+    switch (handle.fate) {
+      case 'create': return ktUtils.applyFun('CreateableStorageKey', [quote(handle.id)]);
```
Comment (mmandlis) [inline src/tools/plan-generator.ts]: shouldn't you be checking for long running arc here? `create` handles of long running arcs should have keys assigned by the recipe2plan, not created by allocator.

Comment (alxrsngrtn) [reply src/tools/plan-generator.ts]: I'm pretty sure we throw an error earlier in the process -- before it reaches generation. 
I can add a `throw` here, or double check (i.e. add unit tests) to ensure that this is the case.

File: src/tools/tests/goldens/WriterReaderExample.kt (hunk 1 above)
Comment (alxrsngrtn) [reply src/tools/tests/goldens/WriterReaderExample.kt]: I'm curious about this, too. I'm running the bazel rule to generate this, and I keep getting something different than what I get from calling the js directly.

Comment (alxrsngrtn) [reply src/tools/tests/goldens/WriterReaderExample.kt]: ts*

File: src/tools/plan-generator.ts (hunk 2 above)
Comment (alxrsngrtn) [reply src/tools/plan-generator.ts]: Notes from meeting
-  check if the storage key is null, then we create the `CreateableStorageKey`. if it's not null, then we do the default.

File: src/tools/tests/goldens/WriterReaderExample.kt (hunk 1 above)
Comment (alxrsngrtn) [reply src/tools/tests/goldens/WriterReaderExample.kt]: Resolved in previous PR.

File: src/tools/plan-generator.ts [hunk 3]
```diff
@@ -109,9 +109,13 @@ export class PlanGenerator {
     }
   }
 
-  /** Generates a Kotlin `StorageKey` from a StorageKey. */
-  createStorageKey(storageKey: StorageKey | undefined): string {
-    return `StorageKeyParser.parse("${(storageKey || '').toString()}")`;
+  /** Generates a Kotlin `StorageKey` from a recipe Handle. */
+  createStorageKey(handle: Handle): string {
+    const storageKey = handle.storageKey;
+    if (!storageKey && handle.fate === 'create') {
```
Comment (mmandlis) [inline src/tools/plan-generator.ts]: should probably assert, if there is no storageKey and fate isn't `create`

File: src/tools/plan-generator.ts [hunk 4]
```diff
@@ -109,9 +109,13 @@ export class PlanGenerator {
...
+      return ktUtils.applyFun('CreateableStorageKey', [quote(handle.id)]);
+    }
+    return ktUtils.applyFun('StorageKeyParser.parse', [(storageKey || '').toString()]);
```
Comment (mmandlis) [inline src/tools/plan-generator.ts]: `storageKey || ''` - the ||'' isn't needed,  the storageKey must not be empty at this point.

nitnit - the local variable storageKey seems redundant, can just use handle.storageKey, i think

**Pull request merged**